src/modules/CurrencyModule.cpp

```diff
@@ -1,46 +1,350 @@
 
-	Module::OnMessage(client, message);
+CurrencyModule::CurrencyModule(UmikoBot* client)
+	: Module("currency", true) {
+
+	m_timer.setInterval(84600 * 1000); //!24hrs = 84600 seconds
+	QObject::connect(&m_timer, &QTimer::timeout, [this, client]() {
+		
+		//!Clear the daily bonus for everyone
+		for (auto it = m_settings.begin(); it != m_settings.end(); ++it) {
+			it->isDailyClaimed = false;
+		}
+
+		if (!isRandomGiveAwayDone) {
+			client->createMessage(giveawayChannelId, "@everyone Today's freebie expires in **10 seconds**. `!claim` it now!");
+
+			allowGiveaway = true;
+
+			QTimer timer;
+			timer.setInterval(10 * 1000);
+			QObject::connect(&timer, &QTimer::timeout, 
+				[this, client]() {
+					if(isRandomGiveAwayDone)
+						isRandomGiveAwayDone = false;
+
+				});
+		}
+	});
+
+	m_timer.start();
+
+	RegisterCommand(Commands::CURRENCY_WALLET, "wallet", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel) {
+		QStringList args = message.content().split(' ');
+		GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+		QString prefix = setting->prefix;
+
+		if (args.first() != prefix + "wallet")
+			return;
+		if (args.size() > 1) {
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+		if (args.size() == 1) {
+			Discord::Embed embed;
+			embed.setColor(11777216);
+			QString creditScore = QString::number(m_settings[setting->id].currency);
+			QString desc = "**Current Credits: ** `" + creditScore + "` **" + currencySymbol + "** (" + currencyName +")";
+			embed.setTitle("Your Wallet");
+			embed.setDescription(desc);
+			client.createMessage(message.channelId(), embed);
+		}
+		});
+
+	RegisterCommand(Commands::CURRENCY_DAILY, "daily", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel) {
+
+		QStringList args = message.content().split(' ');
+		GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+		QString prefix = setting->prefix;
+
+
+		if (args.first() != prefix + "daily")
+			return;
+		if (args.size() > 1) {
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+		if (m_settings[setting->id].isDailyClaimed) {
+
+			int remainingTime = m_timer.remainingTime();
+
+			QTime conv(0, 0, 0);
+			conv = conv.addMSecs(remainingTime);
+			QString time = QString::number(conv.hour()) + "(hrs) " + QString::number(conv.minute()) + "(mins) " + QString::number(conv.second()) + "(secs)";
+			QString desc = "**You have already claimed your daily credits.**\nCome back after `" + time + "` to get more rich!";
+
+			client.createMessage(message.channelId(), desc);
+		}
+		else {
+			m_settings[setting->id].isDailyClaimed = true;
+			m_settings[setting->id].currency += 100;
+
+			client.createMessage(message.channelId(), "**You now have 100 more " + currencyName + " in your wallet!**");
```

Comment:
Can we make the daily amount variable, and changeable with a command, pretty please? So we don't have to change the source code if we want to balance it in the future.

Reply:
sure